File: Miscellaneous/228. Summary Ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> ans;
        for(int i=0;i<nums.size();i++){
            int j=i;
            while(i+1<nums.size()&&nums[i+1]==nums[i]+1){
                i++;
            }
            if(j==i)    ans.push_back(to_string(nums[i]));
            else        ans.push_back(to_string(nums[j])+"->"+to_string(nums[i]));
        }
        return ans;
    }
};

File: 228-summary-ranges/228-summary-ranges.go
func summaryRanges(nums []int) []string {
    var ans []string;
    if len(nums)==0{
        return ans;
    }
    
    first:=nums[0];
    for i:=1;i<len(nums);i++{
        if nums[i]!=(nums[i-1]+1){
            if(first!=nums[i-1]){
                ans=append(ans,strconv.Itoa(first)+"->"+strconv.Itoa(nums[i-1]));
            }else{
                ans=append(ans,strconv.Itoa(first));
            }
            first=nums[i];
        }
    }
    if(first!=nums[len(nums)-1]){
        ans=append(ans,strconv.Itoa(first)+"->"+strconv.Itoa(nums[len(nums)-1]));
    }else{
        ans=append(ans,strconv.Itoa(first));
    }
    
    return ans;
}